ceph/ceph_admin/helper.py

```diff
@@ -290,14 +291,61 @@ def generate_rgw_spec(self, spec):
               rgw_frontend_port: 8080
               rgw_realm: east
               rgw_zone: india
+              rgw_frontend_ssl_certificate: create-cert | <contents of crt>
         Returns:
             service_spec
+
+        contents of rgw_spec.yaml file
+
+            service_type: rgw
+            service_id: rgw.india
+            placement:
+              hosts:
+                - node5
+            spec:
+              ssl: true
+              rgw_frontend_ssl_certificate: |
+                -----BEGIN PRIVATE KEY------
+                ...
         """
         template = self._get_template("rgw")
         node_names = spec["placement"].pop("nodes", None)
         if node_names:
             spec["placement"]["hosts"] = self.get_hostnames(node_names)
 
+        # ToDo: This works for only one host. Not sure, how cephadm handles SSL
+        #       certificate for multiple hosts.
+        if spec["spec"].get("rgw_frontend_ssl_certificate") == "create-cert":
+            subject = {"common_name": spec["placement"]["hosts"][0]}
+            cert, key = generate_self_signed_certificate(subject=subject)
+            pem = key + cert
+            cert_value = "|\n" + pem
+            spec["spec"]["rgw_frontend_ssl_certificate"] = "\n    ".join(
+                cert_value.split("\n")
+            )
+
+            LOG.debug(pem)
+
+            # Copy the certificate to all clients
+            clients = self.cluster.get_nodes(role="client")
+
+            # As the tests are executed on the hosts, copying the certs to them also
+            rgws = self.cluster.get_nodes(role="rgw")
+            nodes = clients + rgws
+
+            for node in nodes:
+                cert_file = node.remote_file(
```

Comment:
Does this holds good for all RGW nodes, even In case of it belongs to different RGW service?

Reply:
done